src/app/components/cart/cart-item/cart-item.component.ts

```diff
@@ -1,27 +1,40 @@
-import { Component, Input } from '@angular/core';
-import { IBook } from '../../../models/book.model';
-import { ICart } from '../../../models/cart.model';
-import { CartService } from '../../../services/cart.service';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { ICartItem } from '../../../models/cart.model';
 
 @Component({
   selector: 'app-cart-item',
   templateUrl: './cart-item.component.html',
   styleUrls: ['./cart-item.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CartItemComponent {
-  @Input() booksInCart!: ICart;
+  @Input() cartItem!: ICartItem;
 
-  constructor(private cartService: CartService) {}
+  @Output() increaseQty = new EventEmitter<number>();
 
-  increaseQty(book: IBook): void {
-    this.cartService.addToCart(book);
+  @Output() decreaseQty = new EventEmitter<number>();
+
+  @Output() deleteItem = new EventEmitter<number>();
+
+  constructor() {
+    this.cartItem = {
```

Comment:
Usually nothing should be done in constructor. If you're adding this because of error about empty cartItem, you can remove ! from it

```@Input() cartItem: ICartItem;```